### PR TITLE
add pcap_to_pandas functionality to netml

### PR DESCRIPTION
**Pull request by feamster**
tested this against notebooks that previously used pcap_to_pandas.  have test script, can check it in somewhere if I know where...

### CODE REVIEW

**Comment by feamster** [comment]
@jesteria I've asked @staveesh to review this pull request for us, since he's taking the class as well.

File: src/netml/pparser/parser.py [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
 import pandas as pd
+
 from scapy.all import *
 from scapy.layers.inet import IP, TCP, UDP
 from sklearn.utils import shuffle
 
-from netml.utils.tool import data_info, timing
+import datetime
+import ipaddress
+import netaddr
```
Comment (staveesh) [inline src/netml/pparser/parser.py:20]: Replacement required in `setup.py` file:
```
INSTALL_REQUIRES = [
    'numpy ~= 1.23.3',
    'pandas ~= 1.5.0',
    'pyod ~= 1.0.5',
    'scapy ~= 2.4.5',
    'scikit-learn ~= 1.1.2',
    'netaddr ~= 0.8.0',
]
```

File: src/netml/pparser/parser.py [hunk 2]
```diff
@@ -933,3 +937,117 @@ def label_flows(self, label_file=None, label=0):
...
+                        pkt_dict["port_src"] = pkt[TCP].sport
+                        pkt_dict["protocol"] = 'TCP'
+                    elif UDP in pkt:
+                        pkt_dict["port_dst"] = pkt[UDP].dport
+                        pkt_dict["port_src"] = pkt[UDP].sport
+                        pkt_dict["protocol"] = 'UDP'
+                    elif ICMP in pkt:
+                        pkt_dict["protocol"] = 'ICMP'
+                        
+                    if DNSQR in pkt:
```
Comment (staveesh) [inline src/netml/pparser/parser.py:1002]: Might want to convert byte strings to strings here.

```
                    if DNSQR in pkt:
                        pkt_dict["is_dns"] = True
                        pkt_dict["dns_query"] = str(pkt[DNSQR].qname, 'utf-8')
                    if DNSRR in pkt:
                        pkt_dict["is_dns"] = True
                        pkt_dict["dns_resp"] = str(pkt[DNSRR].rrname, 'utf-8')
```

**Review by staveesh — CHANGES_REQUESTED** [review]
@feamster @jesteria I tested the `pcap_to_pandas` function on a netflix PCAP and it produced a valid DataFrame. Only a couple of minor changes may be needed further:

1. We may want to convert the byte strings `dns_query` and `dns_resp` to strings.
2. We may want to add the package `netaddr` to `setup.py` as I had to install it explicitly to make it work.

**Review by jesteria — APPROVED** [review]
Thanks folks! :+1:

**Pull request merged**